src/main/java/org/folio/gobi/DataSourceResolver.java

```diff
@@ -77,12 +87,41 @@ private CompletableFuture<String> applyXPath(Document doc) {
       }
   }
 
-  private CompletableFuture<?> applyTranslation(Object o) {
-    if (translation != null) {
-      return translation.apply(o == null ? null : o.toString());
-    } else {
+  public CompletableFuture<?> applyTranslation(Object o) {
+    if (translation == null) {
       return CompletableFuture.completedFuture(o);
     }
+
+    String s = o == null ? null : o.toString();
+    return switch (translation) {
+      case LOOKUP_CONTRIBUTOR_NAME_TYPE_ID -> lookupService.lookupContributorNameTypeId(s);
```

Comment:
I strongly suggest to use ENUM Instead of having variable with same name and remove prefix. it will be more type safe and make it flexible to use. Lookup.CONTRIBUTOR_NAME_TYPE_ID. Of course if it does not break other places and ended up leading much effort

Reply:
I don't get your comment

Reply:
I mean To create ENUM LookUp class inside of Translation class in that wha we eliminate prefix. LookUp_CONTRIBUTOR_NAME_TYPE_ID -> CONTRIBUTOR_NAME_TYPE_ID. having prefix in variables are not good practise. Never mind seems it need extra effort to change ramls. Resolving comment